module/state_synchronization/indexer/persister_test.go

```diff
@@ -0,0 +1,278 @@
+			TransactionID: unittest.IdentifierFixture(),
+			ErrorMessage:  "expected test error",
+			Index:         uint32(i),
+			ExecutorID:    executorID,
+		}
+	}
+	err = p.inMemoryTxResultErrMsg.Store(p.header.ID(), txResultErrMsgs)
+	p.Require().NoError(err)
+}
+
+func (p *PersisterSuite) TestPersister_AddToBatch_WithData() {
+	storedRegisters := make([]flow.RegisterEntry, 0)
+	storedEvents := make([]flow.EventsList, 0)
+	storedCollections := make([]flow.LightCollection, 0)
+	storedTransactions := make([]flow.TransactionBody, 0)
+	storedResults := make([]flow.LightTransactionResult, 0)
+	storedTxResultErrMsgs := make([]flow.TransactionResultErrorMessage, 0)
+
+	p.registers.On("Store", mock.Anything, p.header.Height).Run(func(args mock.Arguments) {
+		sr, ok := args.Get(0).(flow.RegisterEntries)
+		p.Require().True(ok)
+		storedRegisters = sr
+	}).Return(nil)
+
+	p.events.On("BatchStore", p.executionResult.BlockID, mock.Anything, p.batch).Run(func(args mock.Arguments) {
+		se, ok := args.Get(1).([]flow.EventsList)
+		p.Require().True(ok)
+		storedEvents = se
+	}).Return(nil)
+
+	p.collections.On("BatchStoreLightAndIndexByTransaction", mock.Anything, p.batch).Run(func(args mock.Arguments) {
+		collection, ok := args.Get(0).(*flow.LightCollection)
+		p.Require().True(ok)
+		storedCollections = append(storedCollections, *collection)
+	}).Return(nil)
+
+	p.transactions.On("BatchStore", mock.Anything, p.batch).Run(func(args mock.Arguments) {
+		transaction, ok := args.Get(0).(*flow.TransactionBody)
+		p.Require().True(ok)
+		storedTransactions = append(storedTransactions, *transaction)
+	}).Return(nil)
+
+	p.results.On("BatchStore", p.executionResult.BlockID, mock.Anything, p.batch).Run(func(args mock.Arguments) {
+		sr, ok := args.Get(1).([]flow.LightTransactionResult)
+		p.Require().True(ok)
+		storedResults = sr
+	}).Return(nil)
+
+	p.txResultErrMsg.On("BatchStore", p.header.ID(), mock.Anything, p.batch).Run(func(args mock.Arguments) {
+		terrm, ok := args.Get(1).([]flow.TransactionResultErrorMessage)
+		p.Require().True(ok)
+		storedTxResultErrMsgs = terrm
+	}).Return(nil)
+
+	err := p.persister.Persist()
+	p.Assert().NoError(err)
+
+	// Verify all mocks were called as expected
+	p.events.AssertExpectations(p.T())
+	p.Assert().ElementsMatch([]flow.EventsList{p.inMemoryEvents.Data()}, storedEvents)
+
+	p.results.AssertExpectations(p.T())
+	p.Assert().ElementsMatch(p.inMemoryResults.Data(), storedResults)
+
+	p.registers.AssertExpectations(p.T())
+	p.Assert().ElementsMatch(p.inMemoryRegisters.Data(), storedRegisters)
+
+	p.collections.AssertExpectations(p.T())
+	p.Assert().ElementsMatch(p.inMemoryCollections.LightCollections(), storedCollections)
+
+	p.transactions.AssertExpectations(p.T())
+	p.Assert().ElementsMatch(p.inMemoryTransactions.Data(), storedTransactions)
+
+	p.txResultErrMsg.AssertExpectations(p.T())
+	p.Assert().ElementsMatch(p.inMemoryTxResultErrMsg.Data(), storedTxResultErrMsgs)
```

Comment:
no need to assert expectations. As noted [yesterday](https://github.com/onflow/flow-go/pull/7415#discussion_r2150355838), when you use mock constructors, the constructor will automatically add the `AssertExpectations` call. if you run it again, you end up running the same checks twice.

Here is the events mock for reference:
https://github.com/onflow/flow-go/blob/856a2e24cb374611ccc4c90eefce5673c87ddabf/storage/mock/events.go#L193-L203

```suggestion
	p.Assert().ElementsMatch([]flow.EventsList{p.inMemoryEvents.Data()}, storedEvents)
	p.Assert().ElementsMatch(p.inMemoryResults.Data(), storedResults)
	p.Assert().ElementsMatch(p.inMemoryRegisters.Data(), storedRegisters)
	p.Assert().ElementsMatch(p.inMemoryCollections.LightCollections(), storedCollections)
	p.Assert().ElementsMatch(p.inMemoryTransactions.Data(), storedTransactions)
	p.Assert().ElementsMatch(p.inMemoryTxResultErrMsg.Data(), storedTxResultErrMsgs)
```